marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/getDocumentForTesting.sjs

```diff
@@ -19,13 +19,83 @@ xdmp.securityAssert('http://marklogic.com/data-hub/privileges/read-mapping', 'ex
 
 const core = require('/data-hub/5/artifacts/core.sjs')
 const ds = require("/data-hub/5/data-services/ds-utils.sjs");
+const xmlToJson = require('./xmlToJsonForMapping.sjs');
 
 var stepName, uri;
 
+function _isSourceJson(format) {
+  return format.toUpperCase() === 'JSON'
+}
+
+function _isValidQName(name) {
+  try {
+    fn.QName('', name);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+function _isObject(value) {
+  // Added key criteria as typeof(value) returned 'object' for some scalar values.
+  return (value && typeof(value) === 'object' && Object.keys(value).length > 0) === true;
+}
+
+function _isAtomic(value) {
+  return !_isObject(value);
+}
+
+function _isArray(value) {
+  // False negatives from Array.isArray(value)
+  return value && value.hasOwnProperty('0');
+}
+
+function _getXPath(leadingPath, nextPart, value, format, isArray) {
```

Comment:
This function could use some docs on it to explain what each of these args means. And then we typically use "build" and "make" for functions that produce new things, whereas "get" means "get me something from this data structure". Also, I believe the goal of this function is to make a "safe" XPath expression, right? So how about "makeSafeXPathExpression" as the function name, and the docs should clarify what "safe" actually means. 
So I think this should be "makeXPathExpression"

Reply:
Went with `makeSafeXPathExpression()` and included function-level documentation.